app/views/puppetclasses/_class_selection.html.erb

```diff
@@ -9,7 +9,7 @@
   <div class="col-md-4 classes">
     <h3><%= _('Included Classes') %></h3>
 
-    <%= hidden_field_tag "#{obj_type(obj)}[puppetclass_ids][]" %>
+    <%= hidden_field_tag("#{obj_type(obj)}[puppetclass_ids][]") if controller_name == 'config_groups' %>
```

Comment:
I figured out why this is needed for all controllers: if you want to remove all classes, you need to send this empty field. It should move into the `if authorized` block blow - even though the permission seems incorrect for config groups, it is the one used by config groups form as well.

Reply:
I did try that first but did not want to put it there because of the permission for hosts.
I'll test with a user without permissions for host and see if it can still work.